rest_server/src/models/marketplace_item.js

```diff
@@ -37,23 +47,28 @@ class MarketplaceItem {
   }
 
   async list(name, author, category, status) {
-    const handler = modelSyncHandler(async (name, author, category, status) => {
-      const filterStatement = {};
-      if (name) {
-        filterStatement.name = name;
-      }
-      if (author) {
-        filterStatement.author = author;
-      }
-      if (category) {
-        filterStatement.category = category;
-      }
-      if (status) {
-        filterStatement.status = status;
-      }
-      const items = await this.orm.findAll({ where: filterStatement });
-      return items;
-    });
+    const handler = modelSyncHandler(
+      async (name, author, category, status, type) => {
+        const filterStatement = {};
+        if (name) {
+          filterStatement.name = name;
+        }
+        if (author) {
+          filterStatement.author = author;
+        }
+        if (category) {
+          filterStatement.category = category;
```

Comment:
categories is array type, the logic might be not suitable for the new type. Maybe you could delete these related queries that not used now. like category and status

Reply:
Done.